# Homework 4

1. Download the 'bk_logit.csv' data file from the course website.  This is a sample of house and apartment sales in New York City between 2003 and 2013.  The label 'bk' indicates whether a sale occurred in Brooklyn.  The features 'age', 'unit_size', and 'single_unit' measure the age of the unit sold in years, the size of the unit sold in square meters, and an indicator of whether the unit sold is a single-family unit.  Estimate a logit model that relates the label to all of the features in the data.  As shown in class, generate the 'Marginal Effects' table to produce results that measure the effect of a change in the feature on the probability that a sale occurred in Brooklyn.  How does age impact the probability of a sale in Brooklyn?  How does unit size impact the probability of a sale in Brooklyn?  How does a being a single unit impact the probability of a sale in Brooklyn?

2.  Work on your proposals.

In [15]:
%pylab inline
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

Populating the interactive namespace from numpy and matplotlib


`%matplotlib` prevents importing * from pylab and numpy


In [16]:
data = pd.read_csv('bk_logit.csv')

In [17]:
mod = smf.logit('bk ~ age + unit_size + single_unit', data).fit()
print mod.summary()

Optimization terminated successfully.
         Current function value: 0.555319
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:                     bk   No. Observations:                30799
Model:                          Logit   Df Residuals:                    30795
Method:                           MLE   Df Model:                            3
Date:                Fri, 09 Oct 2015   Pseudo R-squ.:                  0.1005
Time:                        19:11:12   Log-Likelihood:                -17103.
converged:                       True   LL-Null:                       -19013.
                                        LLR p-value:                     0.000
                  coef    std err          z      P>|z|      [95.0% Conf. Int.]
-------------------------------------------------------------------------------
Intercept      -2.2461      0.049    -46.114      0.000        -2.342    -2.151
age             0.0186    

In [18]:
print mod.get_margeff().summary()

        Logit Marginal Effects       
Dep. Variable:                     bk
Method:                          dydx
At:                           overall
                 dy/dx    std err          z      P>|z|      [95.0% Conf. Int.]
-------------------------------------------------------------------------------
age             0.0035    8.2e-05     42.534      0.000         0.003     0.004
unit_size       0.0004      2e-05     18.723      0.000         0.000     0.000
single_unit    -0.1558      0.005    -29.286      0.000        -0.166    -0.145


Unit age has a slight postive effect on sale probability, with each year of age increasing the probaility by about .35%.  Size has almost no effect on sale probablity.  Size also has a slight positive effect, with every added square meter increasing probability by .04% (1% increase per 25 square meters).  If a house is single family it has about 15% lower chance to sell.  